src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -1120,4 +1122,46 @@ ValidMPTIssuance::finalize(
         mptokensCreated_ == 0 && mptokensDeleted_ == 0;
 }
 
+//------------------------------------------------------------------------------
+
+void
+ValidPermissionedDomain::visitEntry(
+    bool,
+    std::shared_ptr<SLE const> const& before,
+    std::shared_ptr<SLE const> const& after)
+{
+    if (after->getType() != ltPERMISSIONED_DOMAIN)
+        return;
+    credentialsSize_ = after->getFieldArray(sfAcceptedCredentials).size();
+}
+
+bool
+ValidPermissionedDomain::finalize(
```

Comment:
Thoughts on adding this for `DepositPreauth` as well, while we're here (gated by the PD amendment)?

Reply:
I would suggest a separate PR